controllers/extension-certificate-service/pkg/certservice/actuator.go

```diff
@@ -0,0 +1,262 @@
+	}
+
+	var dns []interface{}
+	for _, route53Provider := range config.Providers.Route53 {
+		route53values := createDNSProviderValuesForDomain(&route53Provider, *shootDomain)
+		dns = append(dns, route53values...)
+	}
+	for _, cloudDNSProvider := range config.Providers.CloudDNS {
+		cloudDNSValues := createDNSProviderValuesForDomain(&cloudDNSProvider, *shootDomain)
+		dns = append(dns, cloudDNSValues...)
+	}
+
+	certConfig := secrets.CertificateSecretConfig{
+		Name:       certBrokerName,
+		CommonName: certBrokerCN,
+	}
+
+	shootKubeconfig, err := util.GetOrCreateShootKubeconfig(ctx, a.client, certConfig, namespace)
+	if err != nil {
+		return err
+	}
+
+	shootKubeconfigChecksum := util.ComputeSecretCheckSum(shootKubeconfig.Data)
+
+	certBrokerConfig := map[string]interface{}{
+		"replicas": util.GetReplicaCount(shoot, 1),
+		"certbroker": map[string]interface{}{
+			"targetClusterSecret": shootKubeconfig.GetName(),
+		},
+		"certmanager": map[string]interface{}{
+			"clusterissuer": config.ACME.IssuerName,
+			"dns":           dns,
+		},
+		"podAnnotations": map[string]interface{}{
+			"checksum/secret-cert-broker": shootKubeconfigChecksum,
+		},
+	}
+
+	certBrokerConfig, err = imagevector.ImageVector.InjectImages(certBrokerConfig, "", "", certBrokerName)
+	if err != nil {
+		return fmt.Errorf("failed to find image version for %s: %v", certBrokerName, err)
+	}
+
+	applier, err := kubernetes.NewChartApplierForConfig(a.config)
+	if err != nil {
+		return fmt.Errorf("failed to create chart applier: %v", err)
+	}
+
+	a.logger.Info("Component is being applied", "component", "cert-broker", "namespace", namespace)
+	return applier.ApplyChartInNamespace(
+		ctx,
+		certBrokerChartPath,
+		namespace,
+		certBrokerName,
+		certBrokerConfig,
+		nil,
+	)
+}
+
+func (a *actuator) deleteCertBroker(ctx context.Context, namespace string) error {
+	chartRenderer, err := chartrenderer.NewForConfig(a.config)
+	if err != nil {
+		return err
+	}
+	configValues := make(map[string]interface{})
+	release, err := chartRenderer.Render(
+		certBrokerChartPath,
+		certBrokerName,
+		namespace,
+		configValues)
+	if err != nil {
+		return fmt.Errorf("failed to render chart: %v", err)
+	}
+
+	var (
+		manifest   = release.Manifest()
+		decoder    = yaml.NewYAMLOrJSONDecoder(bytes.NewReader(manifest), 1024)
+		decodedObj map[string]interface{}
+	)
+
+	a.logger.Info("Component is being deleted", "component", "cert-broker", "namespace", namespace)
+	for err = decoder.Decode(&decodedObj); err == nil; err = decoder.Decode(&decodedObj) {
+		if decodedObj == nil {
+			continue
+		}
+
+		obj := unstructured.Unstructured{Object: decodedObj}
+		decodedObj = nil
+
+		// only delete namepaced resources
+		if len(obj.GetNamespace()) > 0 {
+			if err := a.client.Delete(ctx, &obj); err != nil && !apierrors.IsNotFound(err) {
+				return err
+			}
+		}
+
+	}
+	return nil
+}
+
+func createDNSProviderValuesForDomain(config configv1alpha1.DNSProviderConfig, shootDomain string) []interface{} {
```

Comment:
How about putting these functions that generate chart values into an `internal` package and then test them there?